### ISSUE TITLE
Bug: Dynamic embedding is not included in model.trainable_variables.

### ISSUE THREAD

**Issue by wuxianxingkong** [tensorflow/recommenders-addons#41]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): TF on Yarn.
- TensorFlow version and how it was installed (source or binary): 2.4.1
- TensorFlow-Recommenders-Addons version and how it was installed (source or binary):binary
- Python version: 3.6
- Is GPU used? (yes/no):no

**Describe the bug**

Using example: 
https://github.com/tensorflow/recommenders-addons/blob/master/docs/tutorials/dynamic_embedding_tutorial.ipynb

After running, I print all trainable_variables in model:
self.user_embeddings and self.movie_embeddings is not included in model.trainable_variables which means tape.gradient(loss, model.trainable_variables) will not apply on these embeddings.

**Code to reproduce the issue**

As decribe above.

**Comment by rhdong** [comment]
Hi @wuxianxingkong, thank you for feedback, we will fix it as soon as possible.

**Comment by wuxianxingkong** [comment]
Thanks for your reply. I am looking forward to using this new feature!

**Comment by rhdong** [comment]
Hi @wuxianxingkong , We have fixed this doc issue， plz refer to this [PR](https://github.com/tensorflow/recommenders-addons/pull/43), and optimizer should be wrappered by using `tfra.dynamic_embedding.DynamicEmbeddingOptimizer(tf_optimizer)`

**Comment by wuxianxingkong** [comment]
@rhdong I checked the source of DynamicEmbeddingOptimizer and I found if a variable not in model.trainable_variables, optimizer.apply_gradients will not update TrainableWrapper([here](https://github.com/tensorflow/recommenders-addons/blob/master/tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_optimizer.py#L50) ). I update doc [here](https://github.com/tensorflow/recommenders-addons/pull/46), please review it, thanks

**Comment by wuxianxingkong** [comment]
Additional question: Does dynamic embedding support Parameter Server Strategy in Estimator(Not PS Strategy In TF2.x Keras)?  Because the worker memory increase a lot compared to fc.embedding_column(categorical_column_with_hash_bucket(feature_name, bucket_size_from_statistics)) where bucket_size_from_statistics comes from offline `DISTINCT feature` statistics of training data. And ps memory doesn't increase so obviously. I checked the source in tfra.dynamic_embedding.get_variable, and its `partitioner` param doesn't support standard partitioner such as min_max_variable_partitioner which may mean `variable_scope` and  `get_variable` will not do partition for dynamic embedding.

**Comment by wuxianxingkong** [comment]
![origin](https://user-images.githubusercontent.com/4003232/114492466-3cb66a00-9c4b-11eb-9175-664cf50b6c3e.png)

Origin Memory usage.

**Comment by wuxianxingkong** [comment]
![new](https://user-images.githubusercontent.com/4003232/114492679-9cad1080-9c4b-11eb-92e5-486178224b3b.png)
New Memory usage. Top usage is chief, and memory of worker also increases a lot.

**Comment by rhdong** [comment]
> Additional question: Does dynamic embedding support Parameter Server Strategy(Trainig with CPU)in Estimator?(Not PS Strategy In TF2.x Keras which trains too slow for large model) Because the chief/worker memory increase a lot compared to fc.embedding_column(categorical_column_with_hash_bucket(feature_name, bucket_size_from_statistics)) where bucket_size_from_statistics comes from offline `DISTINCT feature` statistics of training data. And ps memory doesn't increase so obviously. I checked the source in tfra.dynamic_embedding.get_variable, and its `partitioner` param doesn't support standard partitioner such as min_max_variable_partitioner which may mean `variable_scope` and `get_variable` will not do partition for dynamic embedding. And from training logs, "hashpower changed" and "another expansion is on-going" are always printed in chief/all worker logs(not in ps logs) which may caused by hash table needs expansion? But this occurs too frequently.

Firstly, partitioner of de.Variable is different from TF variable's partitioner and maybe you have not set explictly the `devices` of  de.get_variable  that causes the Hashtable run on you worker locally by default.

**Comment by wuxianxingkong** [comment]
@rhdong After setting devices, momory usage has fallen, thanks~

**Comment by rhdong** [comment]
I move this issue to closed status for resolved.
